sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -408,6 +412,10 @@ public String getId() {
      * error.
      */
     public <T> CosmosPagedFlux<T> queryItems(SqlQuerySpec querySpec, CosmosQueryRequestOptions options, Class<T> classType) {
+        if (options == null) {
+            options = new CosmosQueryRequestOptions();
+        }
```

Comment:
I thought that too. We are following this explicit check on api entry point in crud operations. So one thing is consistency, second we have github item #13031 , where we will refactor all apis in one work item, otherwise it will be confusing if we do it for some and leave others